src/Schema/Factories/FieldFactory.php

```diff
@@ -119,331 +69,48 @@ public function handle(FieldValue $fieldValue): array
         // Directives have the first priority for defining a resolver for a field
         /** @var \Nuwave\Lighthouse\Support\Contracts\FieldResolver $resolverDirective */
         if ($resolverDirective = $this->directiveFactory->createSingleDirectiveOfType($fieldDefinitionNode, FieldResolver::class)) {
-            $this->fieldValue = $resolverDirective->resolveField($fieldValue);
+            $fieldValue = $resolverDirective->resolveField($fieldValue);
         } else {
-            $this->fieldValue = $fieldValue->useDefaultResolver();
+            $fieldValue = $fieldValue->useDefaultResolver();
         }
 
-        $fieldMiddleware = $this->passResolverArguments(
-            $this->directiveFactory->createAssociatedDirectivesOfType($fieldDefinitionNode, FieldMiddleware::class)
-        );
-        $this->validationErrorBuffer = (new ErrorBuffer)->setErrorType('validation');
+        $fieldMiddleware = $this->directiveFactory->createAssociatedDirectivesOfType($fieldDefinitionNode, FieldMiddleware::class)
+            // Middleware resolve in reversed order
+            ->push(app(RenameArgsDirective::class))
+            ->push(app(SpreadDirective::class))
+            ->push(app(TransformArgsDirective::class))
+            ->push(app(ValidateDirective::class))
+            ->push(app(SanitizeDirective::class));
```

Comment:
So if I understand correctly everything that either checks or manipulates the incoming request is now done through directives instead of it being hardcoded here in the FieldFactory? Where every directive here handles directives of their type?

Reply:
You got it, that's the idea. That achieves a few things:

- ensure that directives have the power to do most transformations
- keep FieldFactory simple
- allow adding/removing/reordering default field middleware. I plan to expose this capability in a future PR

Reply:
👍 🎉